src/errors/ErrorMessages.ts

```diff
@@ -32,4 +32,5 @@ export enum ERROR_MESSAGES {
     WITHDRAWAL_WAS_NOT_REQUESTED = "Stake withdrawal was not requested",
     STAKE_WAS_NOT_PUT = "Stake was not put",
     INSUFFICIENT_BALANCE = "Signer has insufficient balance",
+    LOGIN_NOT_ENOUGHT_TOKENS = "You are not able to login because you do not have any tokens",
```

Comment:
Why not to use `INSUFFICIENT_BALANCE`? If needed you can extend it with failed operation name

Reply:
I thought it is more to actions like staking. I can change this and i will remove added message.

Reply:
This only checks that balance isn't empty, which still doesn't prevent insufficient funds error. @jrhender wouldn't it make sense to verify balance sufficiency for invoking `Operator` methods cost in did-reg?

Reply:
@JGiter I don't understand, isn't there a check for the required funds amount in this PR? E.g.
```
private async checkBalanceForConnection(): Promise<void> {
        const gasPrice = await this._provider.getGasPrice();
        const gas = await this._provider.estimateGas(this._transactionOverrides);
        const balance =
            (await this._signer?.provider?.getBalance(await this._signer?.getAddress())) || BigNumber.from(0);
        // multiplier 2 chosen arbitrarily because it is not known how reasonably to choose it
        const fee = gasPrice.mul(gas).mul(2);
        if (balance.lte(fee)) {
            throw new Error(ERROR_MESSAGES.INSUFFICIENT_BALANCE);
        }
    }
```
It might make sense to do it in DID registry. Do you mean having a method like `checkForSufficientBalance()` on the `Operator`?

Reply:
This method can't check transaction sent by did-registry. Possible approach to this is to check balance in `Operator._sendTransaction()` and throw specific error which would caught by iam-lib and propagated to Switchboard

Reply:
Ah yes, I see what you are saying. Sure, I can make a story for this.

Reply:
https://energyweb.atlassian.net/browse/EDR-77